components/table/TaskTable.tsx

```diff
@@ -0,0 +1,157 @@
+import { Button } from '@chakra-ui/button';
+import { Box, Flex, Heading } from '@chakra-ui/layout';
+import {
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+    Modal,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+    useDisclosure,
+} from '@chakra-ui/react';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/table';
+import React, { useState } from 'react';
+import ErrorAlert from '../common/ErrorAlert';
+import { ProjectDTO, TaskDTO } from '@/lib/types/dto';
+import { postTask } from '@/lib/utils/apiRequests';
+
+type TaskFields = Partial<TaskDTO> & { project: ProjectDTO; status: 'ACTIVE' };
+
+const fieldsToTask = (fields: TaskFields): TaskDTO => {
+    const { name, status } = fields;
+    if (name && status) {
+        const t: TaskDTO = {
+            ...fields,
+            name,
+            status,
+        };
+        return t;
+    } else {
+        throw 'Invalid task form: missing required fields';
+    }
+};
+
+interface TaskTableProps {
+    project: ProjectDTO;
+    tasks: TaskDTO[];
+}
+
+function TaskTable({ project, tasks }: TaskTableProps): JSX.Element {
+    const [taskFields, setTaskFields] = useState<TaskFields>({
```

Comment:
The idiomatic way to handle forms like these is to use a state with nullable fields representing a form that is still being filled and validate upon submit